mm2src/coins/nft/nft_structs.rs

```diff
@@ -128,14 +132,57 @@ impl fmt::Display for ContractType {
     }
 }
 
+/// `UriMeta` structure is the object which we create from `token_uri` and `metadata`.
+///
+/// `token_uri` and `metadata` usually contain either `image` or `image_url` with image url.
+/// But most often nft creators use only `image` name for this value (from my observation),
+/// less often they use both parameters with the same url.
+///
+/// I suspect this is because some APIs only look for one of these image url names, so nft creators try to satisfy all sides.
+/// In any case, since there is no clear standard, we have to look for both options,
+/// when we build `UriMeta` from `token_uri` or `metadata`.
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub(crate) struct UriMeta {
-    pub(crate) image: Option<String>,
-    #[serde(rename(deserialize = "name"))]
+    #[serde(rename = "image")]
+    pub(crate) raw_image_url: Option<String>,
+    pub(crate) image_url: Option<String>,
+    #[serde(rename = "name")]
     pub(crate) token_name: Option<String>,
-    description: Option<String>,
-    attributes: Option<Json>,
-    animation_url: Option<String>,
+    pub(crate) description: Option<String>,
+    pub(crate) attributes: Option<Json>,
+    pub(crate) animation_url: Option<String>,
+    pub(crate) external_url: Option<String>,
+    pub(crate) image_details: Option<Json>,
+}
+
+impl UriMeta {
+    /// `merge_from` function doesnt change `raw_image_url` field.
+    /// It tries to update `image_url` field instead, if it is None.
+    /// As `image` is the original name of `raw_image_url` field in data from `token_uri` or `metadata`,
+    /// try to find **Some()** in this field first.
+    pub(crate) fn merge_from(&mut self, other: UriMeta) {
```

Comment:
I don't like the `merge_from` name, a name like `fill_missing_from` or `try_to_fill_missing_fields_from`/`try_to_fill_none_fields_from` can be more expressive. I like `try_to_fill_missing_fields_from` it would be used below and would get the message across without having to check the `merge_from` function internals.
```rust
uri_meta.try_to_fill_missing_fields_from(meta_from_meta)
```